### PR TITLE
[testing] add unit tests for SliderQuestion

### PR DESCRIPTION
**Pull request by jonahwilliams**
- Adds type declarations for `SliderQuestion` constructor
- Documents odd behavior of value + 1 on slider.
- Add missing const (we should turn the lint on for this later)

### CODE REVIEW

File: test/ui/widgets/questions_test.dart [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    );
+
+    expect(find.text('subtitle'), findsOneWidget);
+    expect(find.text('title'), findsOneWidget);
+
+    // Includes a SimpleSlider
+    expect(find.byType(SimpleSlider), findsOneWidget);
+
+    // current slider value + 1
+    expect(find.text('50'), findsOneWidget);
```
Comment (jonahwilliams) [inline test/ui/widgets/questions_test.dart:50]: Not sure if this is a bug or not

Comment (jonahwilliams) [reply test/ui/widgets/questions_test.dart:50]: (see initialValue selected as 49)

Comment (dnfield) [reply test/ui/widgets/questions_test.dart:50]: I suspect this is a bug - @joshua-s - what will the expectation be in terms of what gets sent to the backend?

Comment (clocksmith) [reply test/ui/widgets/questions_test.dart:50]: The displayed min and max for the slider is 1 and 5, but it comes from the json as 0 and 4

Comment (jonahwilliams) [reply test/ui/widgets/questions_test.dart:50]: Ahh, so the display needs to be 1 indexed. That makes sense  - a good thing to test :)

Comment (ferndot) [reply test/ui/widgets/questions_test.dart:50]: The original implementation had zero-indexed the slider values. But this does add complexity and is not really gaining us anything here. Feel free to change that. The backend does not depend upon it yet.

Comment (clocksmith) [reply test/ui/widgets/questions_test.dart:50]: I think this test is good, and once the slider data models are updated to be one-index, this test can be updated

**Review by clocksmith — APPROVED** [review]

**Pull request merged**